include/tvm/runtime/device_api.h

```diff
@@ -55,11 +55,13 @@ class DeviceAPI {
   /*!
    * \brief Allocate a data space on device.
    * \param ctx The device context to perform operation.
-   * \param size The size of the memory
+   * \param type The type of elements.
+   * \param nbytes The number of bytes in memory
    * \param alignment The alignment of the memory.
    * \return The allocated device pointer
    */
-  virtual void* AllocDataSpace(TVMContext ctx, size_t size, size_t alignment) = 0;
+  virtual void* AllocDataSpace(TVMContext ctx, TVMType type, size_t nbytes,
```

Comment:
let us put type->type_hint and put it as last parameter. Use comment to say type_hint is only needed by a few backend such as GL

Reply:
Changed.